src/handlers/Firefox110.ts

```diff
@@ -0,0 +1,1041 @@
+
+		// This may throw.
+		sendingRtpParameters.codecs =
+			ortc.reduceCodecs(sendingRtpParameters.codecs, codec);
+
+		const sendingRemoteRtpParameters =
+			utils.clone(this._sendingRemoteRtpParametersByKind![track.kind], {});
+
+		// This may throw.
+		sendingRemoteRtpParameters.codecs =
+			ortc.reduceCodecs(sendingRemoteRtpParameters.codecs, codec);
+
+		// NOTE: Firefox fails sometimes to properly anticipate the closed media
+		// section that it should use, so don't reuse closed media sections.
+		//   https://github.com/versatica/mediasoup-client/issues/104
+		//
+		// const mediaSectionIdx = this._remoteSdp!.getNextMediaSectionIdx();
+		const transceiver = this._pc.addTransceiver(
+			track, {
+				direction     : 'sendonly',
+				streams       : [ this._sendStream ],
+				sendEncodings : encodings
+			});
+
+		const offer = await this._pc.createOffer();
+		let localSdpObject = sdpTransform.parse(offer.sdp);
+
+		// In Firefox use DTLS role client even if we are the "offerer" since
+		// Firefox does not respect ICE-Lite.
+		if (!this._transportReady)
+			await this.setupTransport({ localDtlsRole: 'client', localSdpObject });
+
+		logger.debug(
+			'send() | calling pc.setLocalDescription() [offer:%o]',
+			offer);
+
+		await this._pc.setLocalDescription(offer);
+
+		// We can now get the transceiver.mid.
+		const localId = transceiver.mid;
+
+		// Set MID.
+		sendingRtpParameters.mid = localId;
+
+		localSdpObject = sdpTransform.parse(this._pc.localDescription.sdp);
+
+		const offerMediaObject = localSdpObject.media[localSdpObject.media.length - 1];
+
+		// Set RTCP CNAME.
+		sendingRtpParameters.rtcp.cname =
+			sdpCommonUtils.getCname({ offerMediaObject });
+
+		// Set RTP encodings by parsing the SDP offer if no encodings are given.
+		if (!encodings)
+		{
+			sendingRtpParameters.encodings =
+				sdpUnifiedPlanUtils.getRtpEncodings({ offerMediaObject });
+		}
+		// Set RTP encodings by parsing the SDP offer and complete them with given
+		// one if just a single encoding has been given.
+		else if (encodings.length === 1)
+		{
+			const newEncodings =
+				sdpUnifiedPlanUtils.getRtpEncodings({ offerMediaObject });
+
+			Object.assign(newEncodings[0], encodings[0]);
+
+			sendingRtpParameters.encodings = newEncodings;
+		}
+		// Otherwise if more than 1 encoding are given use them verbatim (but
+		// reverse them back since we reversed them above to satisfy Firefox).
+		else
+		{
+			sendingRtpParameters.encodings = encodings.reverse();
+		}
+
+		// If VP8 or H264 and there is effective simulcast, add scalabilityMode to
+		// each encoding.
+		if (
+			sendingRtpParameters.encodings.length > 1 &&
+			(
+				sendingRtpParameters.codecs[0].mimeType.toLowerCase() === 'video/vp8' ||
+				sendingRtpParameters.codecs[0].mimeType.toLowerCase() === 'video/h264'
+			)
+		)
+		{
+			for (const encoding of sendingRtpParameters.encodings)
+			{
+				encoding.scalabilityMode = 'S1T3';
```

Comment:
In all the other handlers, `Firefox60`, `Chrome74` or `Chrome111`, we retrieve this number from the `temporalLayers`, shouldn't we do the same here ?

Reply:
We have also a recent commit from mediasoup related to this: 
[Fix scalabilityMode: S1T2 is not valid for VP8, use L1T2 instead](https://github.com/versatica/mediasoup-client/pull/247)